addons/survey/models/survey_survey.py

```diff
@@ -181,6 +191,29 @@ def _read_group_states(self, values, domain, order):
         selection = self.env['survey.survey'].fields_get(allfields=['state'])['state']['selection']
         return [s[0] for s in selection]
 
+    @api.onchange('users_login_required', 'certificate')
+    def _onchange_set_certification_give_badge(self):
+        if not self.users_login_required or not self.certificate:
+            self.certification_give_badge = False
+
+    # CRUD
+    @api.model
+    def create(self, vals):
+        if vals.get('certification_give_badge'):
```

Comment:
Just for clarity I would reorder a bit the lines in order to have one call to super
```
if vals.get and not vals.get() -> raise
survey = super()
if vals.get() -> create trigger
return survey
```